lil-lab/recnet
# [recnet-api] arXiv integration

## Description



This PR is mainly working on the feature of arXiv integration, which allows users to paste a random arXiv link and get the metadata directly so they don't need to fill in by themselves. To implement this feature with the scalability of different academic sites(e.g. IEEE, ACM) in the future, this PR introduces the concept of `DigitalLibrary` and mainly uses regular expressions to match the given link.

For more information, please refer to the [design document](https://www.notion.so/Digital-Libraries-Integration-arXiv-26e1673576a84541bb4d789fea2ff518).

This PR also initiates the unit tests of `recnet-api`. I changed some configurations to fix some ci unit tests issues.

## Related Issue





- https://github.com/lil-lab/recnet/issues/256
- https://github.com/lil-lab/recnet/issues/254

## Notes

### How to test locally?

1. Set up your local DB and run the following SQL command to create the arXiv digital library service manually.
```sql
INSERT INTO recnet."DigitalLibrary" ("name", regex, "isVerified", "rank")
VALUES (
  'arXiv', 
  '{"https?:\/\/arxiv\.org\/(abs|pdf)\/(?<id>[0-9.]+(v[0-9]+)?)"}',
  true,
  1
);
```
2. Hit `GET /article` API and put arXiv URL in link query param

- Case 1: Article Link is an arXiv unified link and exists in DB => should return DB value
- Case 2: Article Link is an arXiv unified link and does NOT exist in DB => should fetch arXiv API to get the metadata
  - https://arxiv.org/abs/2409.13688
  - https://arxiv.org/abs/2409.13688v1 (with version)
- Case 3: Article Link is an arXiv pdf link and its unified link exists in DB => should return DB value
  - https://arxiv.org/pdf/2409.13688
  - https://arxiv.org/pdf/2409.13688.pdf
  - https://arxiv.org/pdf/2409.13649v1
  - https://arxiv.org/pdf/2409.13649v1.pdf
- Case 4: Article Link is NOT an arXiv link and exists in DB => should return DB value
- Case 5: Article Link is NOT an arXiv lin and does NOT exist in DB => should return null

## TODO

- [x] Revise Notion document
- [ ] Insert to arXiv data row to db manually
